crates/node_binding/src/plugins/interceptor.rs

```diff
@@ -219,18 +229,40 @@ define_register!(
   cache = false,
 );
 
+/* Compilation Hooks */
 define_register!(
   RegisterCompilationProcessAssetsTaps,
   tap = CompilationProcessAssetsTap<JsCompilation, Promise<()>> @ CompilationProcessAssetsHook,
   cache = false,
 );
 
+/* NormalModuleFactory Hooks */
 define_register!(
   RegisterNormalModuleFactoryBeforeResolveTaps,
   tap = NormalModuleFactoryBeforeResolveTap<JsBeforeResolveArgs, Promise<JsBeforeResolveOutput>> @ NormalModuleFactoryBeforeResolveHook,
   cache = true,
 );
 
+#[async_trait]
+impl AsyncSeries2<Compilation, CompilationParams> for CompilerThisCompilationTap {
+  async fn run(
+    &self,
+    compilation: &mut Compilation,
+    _: &mut CompilationParams,
+  ) -> rspack_error::Result<()> {
+    // SAFETY: `Compiler` will not be moved, as it's stored on the heap.
```

Comment:
This safety comment I wrote is a little bit long and almost look annoying something. Should be shorten in the future 🥲.